### PR TITLE
Partial Fix for issue #1

### PR DESCRIPTION
**Pull request by RickKimball**
This provides a first stab at a Blue Pill Variant to resolve issue #1

### CODE REVIEW

File: variants/STM32F103C8-BluePill/linker_scripts/gcc/STM32F103C8_FLASH.ld [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
   ******************************************************************************
-  * @file    STM32F103RB_FLASH.h
+  * @file    STM32F103CB_FLASH.h
   * @author  WI6LABS
   * @version V1.0.0
   * @date    22-July-2016
-  * @brief   Linker script for STM32F103RB Device with
-  *          128KByte FLASH, 20KByte RAM
+  * @brief   Linker script for STM32F103CB Device with
+  *          64KByte FLASH, 20KByte RAM
```
Comment (edogaldo) [inline variants/STM32F103C8-BluePill/linker_scripts/gcc/STM32F103C8_FLASH.ld:8]: Rick, STM32F103CB is 128KB as well.

Comment (RickKimball) [reply variants/STM32F103C8-BluePill/linker_scripts/gcc/STM32F103C8_FLASH.ld:8]: True, I should have change STM32F103RB to STM32F103C8 ... bad eyes :)

File: variants/STM32F103C8-BluePill/linker_scripts/gcc/STM32F103C8_FLASH.ld [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+  *
+  ******************************************************************************
+  */
+
+/* Entry Point */
+ENTRY(Reset_Handler)
+
+/* Highest address of the user mode stack */
+_estack = 0x20005000;    /* end of RAM */
```
Comment (edogaldo) [inline variants/STM32F103C8-BluePill/linker_scripts/gcc/STM32F103C8_FLASH.ld:49]: @Rick, if you look at standard STM HAL ld files in: system\Drivers\CMSIS\Device\ST\STM32F1xx\Source\Templates\gcc\linker

_estack is always ODD.

**Comment by RickKimball** [comment]
Yes, it is a problem in many HAL setups.  It has shown to causes problems..

https://my.st.com/public/STe2ecommunities/mcu/Lists/STM32Java/Flat.aspx?RootFolder=https%3a%2f%2fmy%2est%2ecom%2fpublic%2fSTe2ecommunities%2fmcu%2fLists%2fSTM32Java%2fSTM32F405RGTx_FLASH%2eld%20uses%20unaligned%20SP&FolderCTID=0x01200200770978C69A1141439FE559EB459D758000F9A0E3A95BA69146A17C2E80209ADC21&currentviews=113

**Review by rick — APPROVED** [review]

**Comment by edogaldo** [comment]
> Yes, it is a problem in many HAL setups. It has shown to causes problems..

Huh, didn't know, thanks!

File: variants/STM32F103C8-BluePill/linker_scripts/gcc/STM32F103C8_FLASH.ld (hunk 2 above)
Comment (RickKimball) [inline variants/STM32F103C8-BluePill/linker_scripts/gcc/STM32F103C8_FLASH.ld]: http://www.stm32duino.com/viewtopic.php?f=48&t=1413&p=18178#p18178

Comment (RickKimball) [reply variants/STM32F103C8-BluePill/linker_scripts/gcc/STM32F103C8_FLASH.ld:49]: http://www.stm32duino.com/viewtopic.php?f=48&t=1413&p=18178#p18178

Comment (edogaldo) [reply variants/STM32F103C8-BluePill/linker_scripts/gcc/STM32F103C8_FLASH.ld:49]: Seen, thanks!

**Comment by rogerclarkmelbourne** [comment]
Wi6Labs have now done a PR to fix the stack and I think the other issue, so I think this is now superseded.

Note boards.txt will be in a state of flux for a while until we nail down how to add addition (non Nucleo boards)

**Comment by rogerclarkmelbourne** [comment]
Superseded by the code from @danieleff

**Pull request closed without merging**